File: example/src/main/java/eu/darken/mvpbakery/example/screens/text/TextPresenter.java
package eu.darken.mvpbakery.example.screens.text;

import android.support.annotation.Nullable;

import javax.inject.Inject;

import eu.darken.mvpbakery.base.Presenter;
import eu.darken.mvpbakery.injection.ComponentPresenter;


@TextComponent.Scope
public class TextPresenter extends ComponentPresenter<TextPresenter.View, TextComponent> {

    private String text;

    @Inject
    TextPresenter() {
    }

    @Override
    public void onBindChange(@Nullable View view) {
        super.onBindChange(view);
        onView(v -> v.showText(text));
    }

    void onTextInput(String text) {
        this.text = text;
    }

    public interface View extends Presenter.View {
        void showText(String text);
    }
}

File: example/src/main/java/eu/darken/mvpbakery/example/screens/text/TextPresenter.kt
package eu.darken.mvpbakery.example.screens.text

import eu.darken.mvpbakery.base.Presenter
import eu.darken.mvpbakery.injection.ComponentPresenter
import javax.inject.Inject


@TextComponent.Scope
class TextPresenter @Inject
internal constructor() : ComponentPresenter<TextPresenter.View, TextComponent>() {

    private var text: String? = null

    override fun onBindChange(view: View?) {
        super.onBindChange(view)
        withView { v -> v.showText(text) }
        withView { v -> v.showPresenterInfo(this) }
    }

    internal fun onTextInput(text: String) {
        this.text = text
    }

    interface View : Presenter.View {
        fun showText(text: String?)

        fun showPresenterInfo(presenter: TextPresenter)
    }
}
